my-app/src/components/messageList.js

```diff
@@ -1,33 +1,48 @@
 import React, { Component } from 'react';
 import Message from './message';
+import slugify from '../utils/slugify';
+import Header from './Header'
 
 class MessageList extends Component {
+  constructor(props){
+    super(props)
+    const channelName = this.props.data.location.data.channelName;
+    const user = this.props.data.username;
+    const slug = slugify(channelName);
+    this.state = {
+      newMessage: "",
+      channelName: this.props.data.location.data.channelName,
+      user: this.props.data.username,
+      slug: slugify(channelName),
+      socket: new WebSocket(`ws://sky-chat.whitechapeau.com/${slug}/${user}`),
+      messagesList: []
+    };
 
-  state = {
-    newMessage: "",
-    messagesList: [
-      {
-      userName: 'Ralf',
-      timePosted: Date.now(),
-      message: 'here is someone\'s message'
-    },
-      {
-        userName: 'Caroline',
-        timePosted: Date.now(),
-        message: 'here is my message'
-      }
-    ]
-  };
+  }
+  componentDidMount(){
+    this.state.socket.addEventListener('open', (event) => {
+      this.state.socket.send('User has joined the chat!');
```

Comment:
you have the user name in state how about using it here?

this.state.socket.send(`${this.state.user} has joined the chat!`)